### PR TITLE
Spring clean the Terraform for the Sierra adapter

### PR DESCRIPTION
**Pull request by alexwlchan**
Part of https://github.com/wellcomecollection/platform/issues/4346

* Move the ECR modules up to the top level
* Use Fargate Spot for the bib merger, item merger and items to Dynamo apps
* Fix all the Terraform 0.12 deprecations

### CODE REVIEW

File: sierra_adapter/terraform/sierra_progress_reporter/lambdas.tf [hunk 1]
```diff
@@ -18,15 +18,21 @@ module "lambda" {
...
+  action        = "lambda:InvokeFunction"
+  function_name = module.lambda.function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.rule.arn
+}
+
+resource "aws_cloudwatch_event_target" "event_trigger_custom" {
+  rule  = aws_cloudwatch_event_rule.rule.id
+  arn   = module.lambda.arn
+  input = "{}"
```
Comment (jamesgorrie) [inline sierra_adapter/terraform/sierra_progress_reporter/lambdas.tf]: ❓

File: sierra_adapter/terraform/sierra_window_generator/lambda.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  function_name = module.window_generator_lambda.function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.window_generator_rule.arn
+}
+
+resource "aws_cloudwatch_event_target" "event_trigger_custom" {
+  rule  = aws_cloudwatch_event_rule.window_generator_rule.id
+  arn   = module.window_generator_lambda.arn
+  input = "{}"
+}
```
Comment (jamesgorrie) [inline sierra_adapter/terraform/sierra_window_generator/lambda.tf]: Any reason for the filename change?

File: sierra_adapter/terraform/sierra_progress_reporter/lambdas.tf (hunk 1 above)
Comment (alexwlchan) [reply sierra_adapter/terraform/sierra_progress_reporter/lambdas.tf]: It’s ported the old module, and I can’t quite remember what it does?

File: sierra_adapter/terraform/sierra_window_generator/lambda.tf (hunk 2 above)
Comment (alexwlchan) [reply sierra_adapter/terraform/sierra_window_generator/lambda.tf]: It shouldn’t be plural, there’s only one Lambda!

**Pull request merged**